src/main/java/com/icthh/xm/ms/otp/web/rest/OneTimePasswordResource.java

```diff
@@ -1,37 +1,65 @@
 package com.icthh.xm.ms.otp.web.rest;
 
 import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.ImmutableMap;
+import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
+import com.icthh.xm.ms.otp.repository.UaaRepository;
+import com.icthh.xm.ms.otp.service.LoginPageRefreshableConfiguration;
 import com.icthh.xm.ms.otp.service.OneTimePasswordService;
 import com.icthh.xm.ms.otp.service.dto.OneTimePasswordCheckDto;
 import com.icthh.xm.ms.otp.service.dto.OneTimePasswordDto;
 import com.icthh.xm.ms.otp.web.rest.errors.BadRequestAlertException;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
 
 import javax.validation.Valid;
 import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Optional;
+
+import static com.icthh.xm.ms.otp.config.Constants.ACCESS_TOKEN;
+import static com.icthh.xm.ms.otp.config.Constants.MSISDN;
 
 /**
  * REST controller for managing OneTimePassword.
  */
 @RestController
 @RequestMapping("/api")
+@RequiredArgsConstructor
 public class OneTimePasswordResource {
 
     private final Logger log = LoggerFactory.getLogger(OneTimePasswordResource.class);
 
     private static final String ENTITY_NAME = "otpOneTimePassword";
-
     private final OneTimePasswordService oneTimePasswordService;
+    private final LoginPageRefreshableConfiguration loginPageRefreshableConfiguration;
+    private final UaaRepository uaaRepository;
+    private final XmAuthenticationContextHolder authenticationContextHolder;
+
+    @Timed
+    @GetMapping(value = "/login", produces = MediaType.TEXT_HTML_VALUE)
+    public ResponseEntity<String> login() {
 
-    public OneTimePasswordResource(OneTimePasswordService oneTimePasswordService) {
-        this.oneTimePasswordService = oneTimePasswordService;
+        if (loginPageRefreshableConfiguration.getLoginHtmlContent() == null){
+            return ResponseEntity.notFound().build();
+        }
+        log.debug("REST login request");
+        return ResponseEntity.ok(loginPageRefreshableConfiguration.getLoginHtmlContent());
```

Comment:
it's incorrect
you must use method without @ReponseBody (now it inside RestController annotation) and return just template name

Reply:
how come i should return template name if i get html string  from config service ?

Reply:
add implementation of class View that return html from config in render method
and add ViewResolver that return you View by name